src/fhirtypes/StructureDefinition.ts

```diff
@@ -133,12 +134,13 @@ export class StructureDefinition {
     let lastMatchId = '';
     for (; i < this.elements.length; i++) {
       const currentId = this.elements[i].id;
-      if (element.id.startsWith(`${currentId}.`) || element.id.startsWith(`${currentId}:`)) {
+      if (new RegExp(`${escapeRegExp(currentId)}[.:/]`).test(element.id)) {
         lastMatchId = currentId;
       } else if (
-        (!currentId.startsWith(`${lastMatchId}.`) && !currentId.startsWith(`${lastMatchId}:`)) ||
+        !new RegExp(`${escapeRegExp(lastMatchId)}[./:]`).test(currentId) ||
         // If element is not a slice at this level, and the currentId is a slice, break to add children before slices
-        (element.id.startsWith(`${lastMatchId}.`) && currentId.startsWith(`${lastMatchId}:`))
+        (new RegExp(`${escapeRegExp(lastMatchId)}[.]`).test(element.id) &&
+          new RegExp(`${escapeRegExp(lastMatchId)}[:/]`).test(currentId))
```

Comment:
In order to get these regular expressions to behave the same as `startsWith`, they should have a `^` at the start.

Reply:
Good call, I will add that.

Reply:
@mint-thompson, this should be ready for re-review based on your comment.